live-build/config/hooks/90-linux-migration-artifact.binary

```diff
@@ -150,13 +150,17 @@ if [[ -n "${DELPHIX_SIGNATURE_TOKEN:-}" ]] && [[ -n "${DELPHIX_SIGNATURE_URL:-}"
 	VERSION="5.3"
 	SIGN_URL="$DELPHIX_SIGNATURE_URL/upgrade/keyVersion/$VERSION/sign"
 
-	set -o pipefail
+	# Encode payload for signature request
+	base64 -w 0 hashes >hashes64
+	echo "{\"data\": \"$(cat hashes64)\"}" >hashes64.payload
+
+	# Request signature
 	curl -s -f -H "Content-Type: application/json" \
-		-u "$DELPHIX_SIGNATURE_TOKEN" "$SIGN_URL" -d @- <<-EOF |
-			{"data": "$(base64 -w 0 $DEPOT_DIRECTORY/hashes)"}
-		EOF
-		jq -r .signature |
-		base64 -d >$DEPOT_DIRECTORY/hashes.sig.$VERSION
+		-u "$DELPHIX_SIGNATURE_TOKEN" "$SIGN_URL" -d @hashes64.payload >hashes.response
+
+	# Decode generated signature
+	jq -r .signature <hashes.response >sig.encoded
```

Comment:
I think we can delete all the transient files after we are done so they do not pollute the depot directory.